### PR TITLE
Add warnings build

### PR DESCRIPTION
**Pull request by rpkilby**
- Adds a fail-able warnings build so deprecations can be caught more proactively
- Updates coverage.py (since python 3.2 is deprecated, we can use coverage 4.0+)
- Updates pytz (2015 raises deprecation warnings)
- Reworks travis/test reqs files
  - `requirements/test.txt` installs `requirements/travis.txt` as a common requirements file
  - travis installs common reqs with -U since some modules are pre-installed/outdated (`mock`).

### CODE REVIEW

**Comment by rpkilby** [comment]
:( 

Need to figure out how to customize the script. Maybe env variables aren't available in that section?

**Comment by rpkilby** [comment]
btw - this is broken/WIP.

**Comment by rpkilby** [comment]
This should be good to go now, although there are some remaining bugs to be fixed both internally and externally. eg, handling m2m rel setting and mock's dependence on pbr, which import the deprecated `imp` module.

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ matrix:
       env: DJANGO="django>=1.10.0,<1.11.0" REST_FRAMEWORK="djangorestframework>=3.4,<3.5"
     - python: "3.3"
       env: DJANGO="django>=1.9.0,<1.10.0" REST_FRAMEWORK="djangorestframework>=3.4,<3.5"
+  include:
+    - python: "3.5"
+      env: W='python -W error -m' DJANGO='https://github.com/django/django/archive/master.tar.gz' REST_FRAMEWORK='djangorestframework>=3.4,<3.5'
   allow_failures:
+    - env: W='python -W error -m' DJANGO='https://github.com/django/django/archive/master.tar.gz' REST_FRAMEWORK='djangorestframework>=3.4,<3.5'
```
Comment (kevin-brown) [inline .travis.yml]: You should be able to set the `-W error` flag using the [`PYTHONWARNINGS` environment variable](http://stackoverflow.com/q/14463277/359284#17654868). This way you wouldn't need to set a custom `W` environment variable.

https://docs.python.org/3.4/using/cmdline.html#envvar-PYTHONWARNINGS

Comment (rpkilby) [inline .travis.yml]: Thanks! (and pushed)

Comment (rpkilby) [inline .travis.yml]: Hm. I'm not sure how to get this to work correctly. By specifying the flag in the env vars, it upgrades all deprecation warnings to errors during the build process itself.

Comment (kevin-brown) [inline .travis.yml]: Usually in builds I do we only upgrade `DeprecationWarning` to an error and make `PendingDeprecationWarning` more verbose but not an error.

```
PYTHONWARNINGS="error::Warning,module::PendingDeprecationWarning"
```

With that being said, I believe there is a way to ignore warnings in specific modules using the syntax.

Comment (rpkilby) [inline .travis.yml]: Either way, the root problem is that the warnings are being raised during the VM's initialization, not during testing.

Comment (carltongibson) [inline .travis.yml]: The intent here is to have a failing build in the Travis output — to give us something to look at — without needing to routinely read the build output, which we won't do. 

Thus, perhaps how it was for now?

Comment (rpkilby) [inline .travis.yml]: @carltongibson - should I revert back to cff0b7c?

Comment (rpkilby) [inline .travis.yml]: Done.

File: requirements/test.txt [hunk 2]
```diff
@@ -1,6 +1,3 @@
--r maintainer.txt
-coverage==3.7.1
-django-discover-runner
-mock
-sphinx-autobuild
+-r travis-ci.txt
+django
```
Comment (carltongibson) [inline requirements/test.txt]: This breaks the `tox` config. Can you remove it please. 

(I'll review unifying the travis and tox configs for 1.0)

**Pull request merged**